tests/pytest/test.py

```diff
@@ -1414,12 +1414,19 @@ def testRDBUnboundedDepth(env):
     # concat the string_126 at the end of itself
     json_value = nest_object(3, 5, "__deep_leaf", 420)
     r.expect('JSON.SET', 'doc', '$..__leaf', json_value).ok()
-    
+
     # RDB dump and restore the key 'doc' and check that the key is still valid
     dump = env.execute_command('dump', 'doc', **{NEVER_DECODE: []})
     r.expect('RESTORE', 'doc1', 0, dump).ok()
     r.expect('JSON.GET', 'doc1', '$..__leaf..__deep_leaf').equal('[420]')
 
+def testUnicodeCharacters(env):
+    # Test JSONPath filter with unicode characters
```

Comment:
Should we test JSONPath filter?

Reply:
We can and probably should; it is just that this particular problem wasn't related to the JSON filter at all, and it also was according to the original ticket description. This particular problem happens very early and isn't a part of JSON path filter processing or any actual processing at all. It is only about the deserialisation of a string into a Rust object, so it happens before everything else.